CMakeLists.txt

```diff
@@ -0,0 +1,1045 @@
+#Sixtrack CMAKE build file
+#Making use of https://cmake.org/Wiki/CMakeForFortranExample
+#and the file created for building Merlin.
+
+###################################################################################################
+# General configuration
+###################################################################################################
+
+#Set the cmake version required
+cmake_minimum_required(VERSION 2.6.4 FATAL_ERROR)
+
+# Set the project name and options
+PROJECT(SixTrack)
+ENABLE_LANGUAGE(Fortran)
+
+file(READ ${CMAKE_SOURCE_DIR}/SixTrack/sixtrack.s IFILE)
+STRING(REGEX MATCH ".*data version /'([0-9]+)[.]([0-9]+)[.]([0-9]+)'/" VERSION_STRING ${IFILE} )
+set(VERSION_STRING "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}.${CMAKE_MATCH_3}")
+MESSAGE(STATUS "Sixtrack version: ${VERSION_STRING}")
+MESSAGE(STATUS "Running on: ${CMAKE_SYSTEM_NAME} with ${CMAKE_SYSTEM_PROCESSOR} and generating for ${CMAKE_GENERATOR}")
+
+#Set Sixtrack version info
+set(${PROJECT_NAME}_VERSION "${VERSION_STRING}")
+set(${PROJECT_NAME}_VERSION_MAJOR "${CMAKE_MATCH_1}")
+set(${PROJECT_NAME}_VERSION_MINOR "${CMAKE_MATCH_2}")
+set(${PROJECT_NAME}_VERSION_PATCH "${CMAKE_MATCH_3}")
+
+# force an out of source build to stop clutter
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+	message(STATUS "Sixtrack requires an out-of-source build.")
+	message(STATUS "Please remove these files from ${CMAKE_BINARY_DIR} first:")
+	message(STATUS "CMakeCache.txt")
+	message(STATUS "CMakeFiles")
+	message(STATUS "Once these files are removed, create a separate directory")
+	message(STATUS "and run CMake from there")
+	message(FATAL_ERROR "in-source build detected")
+endif()
+
+# Update the documentation string of CMAKE_BUILD_TYPE for GUIs to include the MPI options
+SET(CMAKE_BUILD_TYPE Release CACHE STRING
```

Comment:
Hmm. This doesn't really work, does it? Well: If I comment out these lines and don't set -DCMAKE_BUILD_TYPE on the command line, it doesn't build. However, I don't see anywhere the special flags for a debug release?

PS: Before modifying the CMakeLists.txt again, please merge from my cmake_six branch...

In make_six, I use for gfortran when debugging:
`-frecord-marker=4 -$myopt -m32 -g -fno-second-underscore -funroll-loops -mfpmath=sse -msse2 -ffpe-trap=invalid,zero,overflow,underflow -fcheck=all -O0 -g`; normal flags are ` -frecord-marker=4 -$myopt -m32 -g -fno-second-underscore -funroll-loops -mfpmath=sse -msse2`
This is really quite efficient for finding bugs!